grpc-native/src/main/java/org/ballerinalang/net/grpc/ServerCall.java

```diff
@@ -149,6 +147,14 @@ public void sendMessage(Message message) {
         }
     }
 
+    public void injectHeaders(HttpHeaders headers) {
+        if (this.outboundMessage != null) {
+            if (headers != null) {
+                headers.forEach(entry -> this.outboundMessage.setHeader(entry.getKey(), entry.getValue()));
```

Comment:
HttpCarbonMessage has an API 
```
setHeaders(HttpHeaders httpHeaders)
```
Which can be used directly if you have exposed it in the outboundMessage. Can consider this as an improvement later

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/291/commits/1057107aba905b7b2e053520d9e0433e6defd5ed